flake.nix

```diff
@@ -36,7 +36,7 @@
     nixpkgs-update.inputs.mmdoc.follows = "empty";
     nixpkgs-update.inputs.treefmt-nix.follows = "treefmt-nix";
     nixpkgs-update.url = "github:nix-community/nixpkgs-update/infra";
-    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable-small";
+    nixpkgs.url = "https://nixos.org/channels/nixos-unstable-small/nixexprs.tar.xz";
```

Comment:
Actually this url is faster if you regularly switch between nixpkgs versions:

```
--- flake.nix
+++ flake.nix
@@ -2,7 +2,7 @@
   description = "A universal nixos installer, just needs ssh access to the target system";
 
   inputs = {
-    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable-small";
+    nixpkgs.url = "git+https://github.com/NixOS/nixpkgs?shallow=1&ref=nixos-unstable-small";
     flake-parts = { url = "github:hercules-ci/flake-parts"; inputs.nixpkgs-lib.follows = "nixpkgs"; };
 
     # used for testing
```

That cache should work both locally and in buildbot. For fresh clones it's almost as fast as downloading the github archive (tested in github actions)

Reply:
IIRC github has asked projects (homebrew and others) to stop using shallow clones because they are expensive to run on the servers.

Reply:
It's only a shallow clone on the first run because than the cache takes over and it's just a handful commit that is fetched (which should be cheaper than the github archive).
Especially if your internet is not great.

Reply:
That isn't what I was referring to, just using it in this repo isn't the issue, other people copying the pattern is.

Reply:
I see the plan is for nix to migrate from `github:` to shallow `git:` so we may as well switch to your suggestion.